### DISCUSSION TITLE
Use or reimplement QT/KDE suite of libraries

### DISCUSSION THREAD

**Discussion by dphaldes** [slint-ui/slint#4329] [Q&A]
I have barely started looking into Slint, and I am curious how much is Slint interoperable with QT. I know Slint uses QT for window management and native theming. Although, it looks like the widgets available in KDE frameworks and Kirigami (QML library for widgets) would need to be reimplemented from scratch. Is this correct ? 

If I were to consider reimplementing the UI libraries to match the style of Kirigami, would it respect QT theme specific properties like spacing, icons, accent colors. (Also, do you think reimplementing UI libs like this is a viable thing to do ?)

**Comment by ogoffart** [comment]
Hi,
I think your analysis is correct.
Slint optionally depends on Qt to render the native style using QStyle.
If you wanted to implement Kirigami it has to be re-done in Slint.

In the master branch, we expose a palette via the global `Palette` from std-widgets.slint  https://slint.dev/snapshots/master/docs/slint/src/language/builtins/globals#palette this lets you access the colors from the style (which is Qt in the case of the qt style)
Spacing is also exposed via the internal StyleMetrics global.
Icons are not exposed yet. (We were considering integrating with the xdg icons, but that hasn't been done yet)

Re-implementing these controls seems like a nice thing to do, although it may be a lot of work.
